compiler/psi/src/org/jetbrains/kotlin/psi/KtTypeReference.kt

```diff
@@ -49,6 +49,8 @@ class KtTypeReference : KtModifierListOwnerStub<KotlinPlaceHolderStub<KtTypeRefe
         return modifierList?.annotationEntries.orEmpty()
     }
 
+    override fun getName() = if (typeElement is KtUserType) (typeElement as KtUserType).referencedName else null
```

Comment:
I'd better use a different scary name like `getNameForAdditionalReceiverLabel`, so we were sure it's not abused anyhow